app/models/spree/calculator/flowcommerce_calculator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Spree
+  class Calculator
+    class FlowcommerceCalculator < Calculator::DefaultTax
+      def self.description
+        'FlowCommerce Calculator'
+      end
+
+      def compute_shipment_or_line_item(item)
+        order = item.order
+
+        if can_calculate_tax?(order)
+          flow_response = get_flow_response(order)
+          tax_for_item(item, flow_response)
+        else
+          prev_tax_amount(item)
+        end
+      end
+      alias compute_shipment compute_shipment_or_line_item
+      alias compute_line_item compute_shipment_or_line_item
+
+      private
+
+      def prev_tax_amount(item)
+        if rate.included_in_price
+          item.included_tax_total
+        else
+          item.additional_tax_total
+        end
+      end
+
+      def can_calculate_tax?(order)
+        return false if order.flow_data.blank?
+        return false if %w[cart address delivery].include?(order.state)
+
+        true
+      end
+
+      def get_flow_response(order)
```

Comment:
@sebastiandl If this method is returning the tax calculation from flow, can we add a better name to the method? `get_flow_reponse(order)` doesn't give any clue about what kind of response it returns. Thanks!

Reply:
Addittionally please address the warnings. Looks like we have some useless warnings on aliases

Reply:
makes sense! I ll update this method's name probably something like `get_flow_allocations(order)` probably is better.